product_lot_sequence/models/stock_move.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def action_show_details(self):
+        """Avoid calling and incrementing the sequence if not needed or already done"""
+        seq_policy = self.env["stock.production.lot"]._get_sequence_policy()
+        if seq_policy in ("product", "global"):
+            # If move is not supposed to assign serial pass empty string for next serial
+            if not self.display_assign_serial:
+                return super(
+                    StockMove, self.with_context(force_next_serial="")
+                ).action_show_details()
```

Comment:
Variable `self` is looked up in local context by `super()`, so you could simplify:
```suggestion
                self = self.with_context(force_next_serial="")
```

Reply:
I didn't want to propagate the context key outside of the super call scope (what would be the case if we alter `self`)